source/frontend/src/features/leases/detail/LeasePages/tenant/AddLeaseTenantContainer.tsx

```diff
@@ -37,8 +45,13 @@ export const AddLeaseTenantContainer: React.FunctionComponent<
     requestName: 'get person by id',
   });
 
-  const setSelectedTenantsWithPersonData = async (tenants?: IContactSearchResult[]) => {
-    const personPersonIdList = getTenantOrganizationPersonList(tenants);
+  const setSelectedTenantsWithPersonData = async (updatedTenants?: IContactSearchResult[]) => {
+    const allExistingTenantIds = tenants.map(t => t.id);
```

Comment:
previously processed tenants should be ignored, which ensures that if the user selected a primary contact it is not overwritten.